Makefile

```diff
@@ -23,3 +23,26 @@ docker-build:
 .PHONY: offsets
 offsets:
 	cd offsets-tracker; OFFSETS_OUTPUT_FILE="../pkg/inject/offset_results.json" go run main.go
+
+.PHONY: install-tools
+install-tools:
+	go install github.com/google/go-licenses@latest
+
+.PHONY: update-licenses
+update-licenses: install-tools
+	rm -rf LICENSES; \
+	go-licenses save ./cli/ --save_path LICENSES; \
+	cp -R ./include/libbpf ./LICENSES
+
+.PHONY: verify-licenses
+verify-licenses: install-tools
+	go-licenses save ./cli --save_path temp; \
+	cp -R ./include/libbpf ./temp; \
+    if diff temp LICENSES > /dev/null; then \
+      echo "Passed"; \
+      rm -rf temp; \
+    else \
+      echo "LICENSES directory must be updated. Run make update-licenses"; \
+      rm -rf temp; \
+      exit 1; \
+    fi; \
```

Comment:
We can simply run `git diff` after `update-licences` in CI to validate if there are no missing changes.

Something like (not tested!):

```make
.PHONY: gitdiff
gitdiff:
          git status
          git diff
          test -z "$(shell git status --porcelain)"
```

We should also have a GitHub workflow to make sure it is up to date.

Reply:
I am not sure if I follow. You can have a commit that bumps the dependency (via dependabot) and then you add commit that updates the licenses (if they changed) or you execute `make update-licences` after adding a new dependency. You can also do both in one commit.

Anyway. I do not have a strong opinion. The good thing about my proposal that it would be also compatible with other "code generation" tools e.g. `go generate`.

Reply:
You're right, I was mixing the generation and verification into one step.  With your method you could bump the dependency, then run `make updated-licenses`, commit all of that, and then run `make verify-licenses`, which would run `update-licenses` and `git-diff`, erroring if there are any changes.

So I think it comes down to how much the failed error messages matters vs compatibility with other tools like `go generate`.

Reply:
Testing this out I have instantly run into `git diff` seeing the `.DS_Store` file on my mac :/

Reply:
> Testing this out I have instantly run into `git diff` seeing the `.DS_Store` file on my mac :/

side note: I encourage you to add `.DS_Store` to you global git ignore 😉 See: https://gist.github.com/subfuzion/db7f57fff2fb6998a16c

Reply:
That is an excellent idea, thanks for sharing; implemented immediately.  But I don't think we can rely on every contributor having done that.

Reply:
Somehow I missed your second comment 👍